org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/linkedediting/LSPLinkedEditingReconcilingStrategy.java

```diff
@@ -248,6 +252,21 @@ String getValueInRange(LinkedEditingRanges ranges, VerifyEvent event, int offset
 		return null;
 	}
 
+	private boolean rangesContainOffset(LinkedEditingRanges ranges, int offset) {
+		try {
+			for (Range range : ranges.getRanges()) {
+				int start = LSPEclipseUtils.toOffset(range.getStart(), fDocument);
+				int end = LSPEclipseUtils.toOffset(range.getEnd(), fDocument);
```

Comment:
if we would move  LSPEclipseUtils.toOffset(range.getEnd(), fDocument) into the if we would benefit from lazy evaluation.
the same would apply to the existing logic in getValueInRange.
Maybe factor out to a method

boolean rangeContainOffset(Range range, int offset) 

and reuse it?

Reply:
I second this. Big fan of lazy evaluation myself :-)

Reply:
Agree